src/details/ArborX_DetailsCartesianGrid.hpp

```diff
@@ -118,6 +121,23 @@ struct CartesianGrid
       m /= _n[d - 1];
       ARBORX_ASSERT(_n[d] < m);
     }
+
+    // Catch a potential loss of precision that may happen in cellBox() and can
+    // lead to wrong results.
+    //
+    // The machine precision by itself is not sufficient. In some experiments
+    // run with a full NGSIM datasets, values below 3 could still produce wrong
+    // results. This may still not be conservative enough, but all runs passed
+    // verification when this warning was not triggered.
+    float constexpr eps = 5 * std::numeric_limits<float>::epsilon();
+    for (int d = 0; d < DIM; ++d)
+    {
+      if (std::abs(_h[d] / min_corner[d]) < eps)
+        std::cerr << "ArborX warning: FDBSCAN-DenseBox algorithm may "
+                     "experience loss of precision, producing wrong results. "
+                     "Please consider switching to FDBSCAN."
+                  << std::endl;
```

Comment:
Fine but the will be the first occurrence of printing to `cerr` or `cout` and I am not looking forward to introducing some mechanism to not print the warnings down the road.